File: javascript/bob/bob.js
var Bob = function() {};

Bob.prototype.hey = function(input) {

  if(input.toUpperCase() == input && input.toLowerCase() != input)
    return 'Whoa, chill out!';

  else if(input.replace(/\s/g, '').length == 0)
    return 'Fine. Be that way!'

  else if(input.indexOf('?') == input.length - 1)
    return 'Sure.';

  else
    return 'Whatever.';

};

module.exports = Bob;

File: scala/bob/src/main/scala/Bob.scala
class Bob {

  def hey(input: String) : String =
    if(input.toUpperCase == input && input.toLowerCase() != input)
      return "Whoa, chill out!"
    else if(input.replaceAll("""\s+""", "").length == 0)
      return "Fine. Be that way!"
    else if(input.takeRight(1) == "?")
      return "Sure."
    else
      return "Whatever."

}
